pkg/quantile/agent.go

```diff
@@ -50,14 +54,24 @@ func (a *Agent) Reset() {
 }
 
 // Insert v into the sketch.
-func (a *Agent) Insert(v float64) {
-	a.Sketch.Basic.Insert(v)
+func (a *Agent) Insert(v float64, sampleRate float64) {
+	k := agentConfig.key(v)
+	if sampleRate == 1 {
+		a.Sketch.Basic.Insert(v)
+		a.Buf = append(a.Buf, k)
 
-	a.Buf = append(a.Buf, agentConfig.key(v))
-	if len(a.Buf) < agentBufCap {
-		return
+		if len(a.Buf) < agentBufCap {
+			return
+		}
+	} else {
+		n := uint(math.Ceil(1 / sampleRate))
+		a.Sketch.Basic.InsertN(v, n)
+		kc := KeyCount{
+			k: k,
+			n: n,
+		}
+		a.CountBuf = append(a.CountBuf, kc)
```

Comment:
Is this case intended to always fall through to `flush`?

Reply:
Yes I think we always want to flush the CountBuf since the counts could be high, whereas above the count is always 1